code/modules/pai/pai.dm

```diff
@@ -191,30 +191,32 @@
 	else
 		. += "Systems nonfunctional."
 
-/mob/living/silicon/pai/handle_atom_del(atom/deleting_atom)
-	if(deleting_atom == hacking_cable)
-		untrack_pai()
-		untrack_thing(hacking_cable)
-		hacking_cable = null
-		SStgui.update_user_uis(src)
-		if(!QDELETED(card))
-			card.update_appearance()
-	if(deleting_atom == atmos_analyzer)
+/mob/living/silicon/pai/Exited(atom/movable/gone, direction)
+	if(gone == atmos_analyzer)
 		atmos_analyzer = null
-	if(deleting_atom == camera)
+	else if(gone == camera)
 		camera = null
-	if(deleting_atom == host_scan)
+	else if(gone == host_scan)
 		host_scan = null
-	if(deleting_atom == internal_gps)
+	else if(gone == internal_gps)
 		internal_gps = null
-	if(deleting_atom == instrument)
+	else if(gone == instrument)
 		instrument = null
-	if(deleting_atom == newscaster)
+	else if(gone == newscaster)
 		newscaster = null
-	if(deleting_atom == signaler)
+	else if(gone == signaler)
 		signaler = null
 	return ..()
 
+/mob/living/silicon/pai/proc/on_hacking_cable_del(atom/source)
+	SIGNAL_HANDLER
+	untrack_pai()
+	untrack_thing(hacking_cable)
+	hacking_cable = null
+	SStgui.update_user_uis(src)
+	if(!QDELETED(card))
+		card.update_appearance()
```

Comment:
pai stuff is SUPER DUPER temperamental. have you tested this one?

Reply:
Yes, and it works just fine.